packages/pxweb2/src/mappers/JsonStat2ResponseMapper.ts

```diff
@@ -189,47 +261,148 @@ function mapDimension(id: string, dimension: any, role: any): Variable | null {
+  };
+}
+
+/**
+ * Maps the notes at value level for the given dimension.
+ *
+ * @param dimension - The dimension object from the JSON-stat 2.0 response.
+ * @param mappedValues - The array of Value objects to map the notes to.
+ */
+function mapValueNotes(dimension: any, mappedValues: Value[]): void {
+  if (
+    dimension.category?.index &&
+    dimension.category.label &&
+    dimension.category.note
+  ) {
+    const noteEntries = Object.entries(dimension.category.note);
+
+    for (const [code] of noteEntries) {
+      if (Object.hasOwn(dimension.category.note, code)) {
+        const noteTexts = dimension.category.note[code];
+
+        for (let i = 0; i < noteTexts.length; i++) {
+          let newNote: Note = {
+            text: noteTexts[i],
+            mandatory: getMandatoryValueNote(dimension.extension, code, i),
+          };
+
+          addNoteToItsValue(code, newNote, mappedValues);
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Returns whether a note at value level is mandatory.
+ *
+ * @param dimensionExtension - The dimension extension object from the JSON-stat 2.0 response.
+ * @param code - The code of the value.
+ * @param noteIndex - The index of the note.
+ * @returns True if the note is mandatory; otherwise, false.
+ */
+function getMandatoryValueNote(
+  dimensionExtension: extension_dimension,
+  code: string,
+  noteIndex: number,
+): boolean {
+  if (dimensionExtension?.categoryNoteMandatory) {
+    if (dimensionExtension.categoryNoteMandatory[code]) {
+      if (dimensionExtension.categoryNoteMandatory[code][noteIndex]) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+/**
+ * Adds a note to its corresponding value.
+ *
+ * @param code - The code of the value.
+ * @param note - The note to add.
+ * @param values - The array of Value objects to add the note to.
+ */
+function addNoteToItsValue(code: string, note: Note, values: Value[]): void {
+  const mappedValue = values.find((v) => v.code === code);
+  if (mappedValue) {
+    if (!mappedValue.notes) {
+      mappedValue.notes = [];
+    }
+    mappedValue.notes?.push(note);
+  }
+}
+
+/**
+ * Maps the value display type for a dimension.
+ *
+ * @param dimensionExtension - The dimension extension object from the JSON-stat 2.0 response.
  * @returns The value display type for the dimension.
  */
-function getValueDisplayType(dimension: any): ValueDisplayType {
-  if (dimension.extension?.show) {
-    if (dimension.extension.show === 'code') {
+function getValueDisplayType(
+  dimensionExtension: extension_dimension,
+): ValueDisplayType {
+  if (dimensionExtension?.show) {
+    if (dimensionExtension.show === 'code') {
       return 'code';
-    } else if (dimension.extension.show === 'value') {
+    } else if (dimensionExtension.show === 'value') {
       return 'value';
-    } else if (dimension.extension.show === 'code_value') {
+    } else if (dimensionExtension.show === 'code_value') {
       return 'code_value';
     }
```

Comment:
More a preference comment, the "else if"s here are not really needed, since there are returns inside of each. Personally I prefer pure ifs, but this is subjective I think.